modules/authglobus/lib/Auth/Source/Globus.php

```diff
@@ -73,36 +74,91 @@ protected function flatten($array, $prefix = '')
-     * Exchange an authorization code for Globus Tokens.
-     *
-     * @param array $contents The contents of the POST request. REF: https://docs.globus.org/api/auth/reference/#authorization_code_grant_preferred
-     * @param string $signature Your Client ID + Secret from the Globus app you've registered.
-     */
-    private function getGlobusTokens($contents, $signature) {
-        $curl = curl_init();
+    * Accepts a field containing a full name, attempts to extract a full name
+    *
+    * @param string $name
+    *
+    * @return an array containing first name and last name
+    */
+    protected function getFullName($name) {
+        $trimmedName = trim($name);
 
-        curl_setopt($curl, CURLOPT_USERAGENT, 'SSPHP Globus'); //umad Globus
-        curl_setopt($curl, CURLOPT_USERPWD, $signature);
-        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
-        curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+        if ($trimmedName && (strpos($trimmedName, ',') !== null || strpos($trimmedName, '\s') !== null)) {
+            $delimiter = strpos($name, ',') !== null ? ',' : '\s';
+            $nameParts = explode($delimiter, $trimmedName, 2);
 
-        curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
-        curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
+            $firstName;
+            $lastName;
 
-        curl_setopt($curl, CURLOPT_URL, self::API_ENDPT . '/oauth2/token');
-        curl_setopt($curl, CURLOPT_POSTFIELDS, OauthUtil::build_http_query($contents));
+            if ($delimiter === '\s') {
+                $firstName = $nameParts[0];
+                $lastName = $nameParts[1];
+            } else {
+                $firstName = $nameParts[1];
+                $lastName = $nameParts[0];
+            }
 
-        $newArr = curl_exec($curl);
-        $outInfo = curl_getinfo($curl);
+            return array(
+                'first_name' => $firstName,
+                'last_name' => $lastName
+            );
+        }
 
-        if (curl_error($curl)) {
-            throw new SimpleSAML_Error_BadRequest('Could not retrieve access token(s) from Globus.');
+        return array(
+            'first_name' => 'UNKNOWN',
+            'last_name' => 'UNKNOWN'
+        );
+    }
+
+    protected function doCurlRequest(
+        $path,
+        $signature = null,
+        $token = null,
+        $qs = null,
+        $contents = null,
+        $method = 'GET'
+    ) {
+        curl_setopt($this->curl, CURLOPT_USERAGENT, 'SSPHP Globus'); //umad Globus
+
+        if ($signature !== null) {
+            curl_setopt($this->curl, CURLOPT_USERPWD, $signature);
         }
+        
+        curl_setopt($this->curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($this->curl, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($this->curl, CURLOPT_RETURNTRANSFER, 1);
 
-        return $newArr;
+        curl_setopt($this->curl, CURLOPT_SSL_VERIFYHOST, false);
```

Comment:
Do we actually not want to verify the SSL?  Since this is communicating with a known service we should probably turn this on in the production version